commit b2d3ea4a730f812b9c0f67a67b6762ce66ddb17c
Author: Eric Dumazet <edumazet@google.com>
Date:   Mon Sep 19 23:39:12 2016 -0400

    tcp: switch back to proper tcp_skb_cb size check in tcp_init()
    
    Revert to the tcp_skb_cb size check that tcp_init() had before commit
    b4772ef879a8 ("net: use common macro for assering skb->cb[] available
    size in protocol families"). As related commit 744d5a3e9fe2 ("net:
    move skb->dropcount to skb->cb[]") explains, the
    sock_skb_cb_check_size() mechanism was added to ensure that there is
    space for dropcount, "for protocol families using it". But TCP is not
    a protocol using dropcount, so tcp_init() doesn't need to provision
    space for dropcount in the skb->cb[], and thus we can revert to the
    older form of the tcp_skb_cb size check. Doing so allows TCP to use 4
    more bytes of the skb->cb[] space.
    
    Fixes: b4772ef879a8 ("net: use common macro for assering skb->cb[] available size in protocol families")
    Signed-off-by: Eric Dumazet <edumazet@google.com>
    Signed-off-by: Soheil Hassas Yeganeh <soheil@google.com>
    Signed-off-by: Neal Cardwell <ncardwell@google.com>
    Signed-off-by: Yuchung Cheng <ycheng@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/ipv4/tcp.c b/net/ipv4/tcp.c
index e79ed17ccfd6..de02fb4b1349 100644
--- a/net/ipv4/tcp.c
+++ b/net/ipv4/tcp.c
@@ -3261,11 +3261,12 @@ static void __init tcp_init_mem(void)
 
 void __init tcp_init(void)
 {
-	unsigned long limit;
 	int max_rshare, max_wshare, cnt;
+	unsigned long limit;
 	unsigned int i;
 
-	sock_skb_cb_check_size(sizeof(struct tcp_skb_cb));
+	BUILD_BUG_ON(sizeof(struct tcp_skb_cb) >
+		     FIELD_SIZEOF(struct sk_buff, cb));
 
 	percpu_counter_init(&tcp_sockets_allocated, 0, GFP_KERNEL);
 	percpu_counter_init(&tcp_orphan_count, 0, GFP_KERNEL);
